apps/backend/src/modules/auth/auth.service.ts

```diff
@@ -1,20 +1,39 @@
+import {
+	type BaseToken,
+	type TokenPayload,
+} from "~/libs/modules/token/token.js";
 import {
 	type UserService,
 	type UserSignUpRequestDto,
 	type UserSignUpResponseDto,
 } from "~/modules/users/users.js";
 
+type Constructor = {
+	tokenService: BaseToken;
+	userService: UserService;
+};
+
 class AuthService {
+	private tokenService: BaseToken;
+
 	private userService: UserService;
 
-	public constructor(userService: UserService) {
+	public constructor({ tokenService, userService }: Constructor) {
+		this.tokenService = tokenService;
 		this.userService = userService;
 	}
 
 	public async signUp(
 		userRequestDto: UserSignUpRequestDto,
 	): Promise<UserSignUpResponseDto> {
-		return await this.userService.create(userRequestDto);
+		const { email, id } = await this.userService.create(userRequestDto);
+		const token = await this.generateToken(id);
```

Comment:
How we generate tokens may differ from the methods where we use them, and the logic here is pretty straightforward, so it's better not to isolate it, just to pass the user ID. If we wanted to isolate it, it would be something more specific, e.g., generateUserAuthToken.
```suggestion
		const { email, id } = await this.userService.create(userRequestDto);
		const token = await this.tokenService.createToken({ userId: user.id });
```